pkg/shell/topnav.jsx

```diff
@@ -212,20 +212,20 @@ export class TopNav extends React.Component {
 
         if (manifest.locales)
             main_menu.push(<DropdownItem key="languages" className="display-language-menu"
-                                         onClick={() => Dialogs.show(<LangModal />)}>
+                                         onClick={() => Dialogs.run(LangModal, {})}>
                 {_("Display language")}
             </DropdownItem>);
 
         if (this.state.showActivePages)
             main_menu.push(
                 <DropdownItem key="frames" id="active-pages" component="button"
-                              onClick={() => Dialogs.show(<ActivePagesDialog frames={this.props.index.frames} />)}>
+                              onClick={() => Dialogs.run(ActivePagesDialog, { frames: this.props.index.frames })}>
                     {_("Active pages")}
                 </DropdownItem>);
 
         main_menu.push(
             <DropdownItem key="creds" id="sshkeys" component="button"
-                          onClick={() => Dialogs.show(<CredentialsModal />)}>
+                          onClick={() => Dialogs.run(CredentialsModal, {})}>
```

Comment:
I guess there is no other way to dynamically pass a promise to a new component but I kinda liked the expectable syntax of JSX. Which reads like "oh this will render `<CredentialsModal \>` as a component.

The second argument `{}` could be optional I'd say.

Reply:
I thought contexts *are* the mechanism to do that?